core/runtime/src/main/java/io/quarkus/artemis/core/runtime/health/ServerLocatorHealthCheck.java

```diff
@@ -15,68 +16,34 @@
 import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
 import org.eclipse.microprofile.health.Readiness;
 
-import io.quarkus.arc.Arc;
-import io.quarkus.arc.InstanceHandle;
 import io.quarkus.artemis.core.runtime.ArtemisBuildTimeConfigs;
 import io.quarkus.artemis.core.runtime.ArtemisRuntimeConfigs;
 import io.quarkus.artemis.core.runtime.ArtemisUtil;
-import io.smallrye.common.annotation.Identifier;
 
 @Readiness
 @ApplicationScoped
 public class ServerLocatorHealthCheck implements HealthCheck {
-    private final HashMap<String, ServerLocator> serverLocators = new HashMap<>();
+    private final Instance<ServerLocator> serverLocators;
+    private final Set<String> serverLocatorNames;
 
     public ServerLocatorHealthCheck(
             ArtemisRuntimeConfigs runtimeConfigs,
             ArtemisBuildTimeConfigs buildTimeConfigs,
-            @SuppressWarnings("CdiInjectionPointsInspection") ArtemisHealthSupport support) {
-        HashSet<String> includedNames = new HashSet<>(support.getConfiguredNames());
-        includedNames.removeAll(support.getExcludedNames());
-        processKnownBeans(runtimeConfigs, includedNames);
-        processArcBeans(runtimeConfigs, buildTimeConfigs);
-    }
-
-    private void processKnownBeans(ArtemisRuntimeConfigs runtimeConfigs, HashSet<String> includedNames) {
-        for (String name : includedNames) {
-            if (runtimeConfigs.configs().get(name).isHealthInclude()) {
-                Annotation identifier;
-                if (ArtemisUtil.isDefault(name)) {
-                    identifier = Default.Literal.INSTANCE;
-                } else {
-                    identifier = Identifier.Literal.of(name);
-                }
-                ServerLocator locator;
-                try (InstanceHandle<ServerLocator> handle = Arc.container().instance(ServerLocator.class, identifier)) {
-                    locator = handle.get();
-                }
-                if (locator != null) {
-                    serverLocators.put(name, locator);
-                }
-            }
-        }
-    }
-
-    private void processArcBeans(ArtemisRuntimeConfigs runtimeConfigs, ArtemisBuildTimeConfigs buildTimeConfigs) {
-        if (runtimeConfigs.getHealthExternalEnabled()) {
-            HashSet<String> namesToIgnore = new HashSet<>(runtimeConfigs.configs().keySet());
-            namesToIgnore.addAll(buildTimeConfigs.configs().keySet());
-            Map<String, ServerLocator> locatorNamesFromArc = ArtemisUtil.extractIdentifiers(ServerLocator.class, namesToIgnore);
-            for (var entry : locatorNamesFromArc.entrySet()) {
-                ServerLocator locator = entry.getValue();
-                if (locator != null) {
-                    serverLocators.put(entry.getKey(), locator);
-                }
-            }
-        }
+            @SuppressWarnings("CdiInjectionPointsInspection") ArtemisHealthSupport support,
```

Comment:
Why `@SuppressWarnings` is needed here? Can't we add `@Inject` in the construtor to make the compiler happy?

Reply:
Nope. This is more of an IntelliJ-thing. For whatever reason, IntelliJ is not able to see the bean. I normally suppress those warnings when I know that the beans are present.